### PR TITLE
[ENH]: Logging with a color formatter

### PR DESCRIPTION
**Pull request by fraimondo**
Taking advantage of the xterm-colors... this PR adds colors to logging messages.

Screenshot 2024-12-05 at 10 47 17

It also allows to set the datalad logger to a different level than the one from junifer, allowing to debug junifer without the million debugging messages from datalad.

### CODE REVIEW

File: junifer/utils/logging.py [hunk 1]
```diff
@@ -297,11 +351,19 @@ def configure_logging(
         #     "%(asctime)s [%(levelname)s] %(message)s "
         #     "(%(filename)s:%(lineno)s)"
         # )
-    formatter = logging.Formatter(fmt=output_format)
+    formatter = ColorFormatter(fmt=output_format)
```
Comment (synchon) [inline junifer/utils/logging.py:354]: What about an option for a `--color` parameter for switching between the formatters? Maybe some remote machine doesn't have xterm support and that might break it?

**Review by synchon — CHANGES_REQUESTED** [review]

File: junifer/utils/logging.py [hunk 2]
```diff
@@ -80,6 +80,59 @@ def __getattr__(self, name: str) -> str:
...
+    Parameters
+    ----------
+    fmt : str
+        The format string for the logging message.
+    datefmt : str, optional
+        The format string for the date.
+
+    """
+
+    BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE = range(8)
```
[the remark this answers is not in the document]
Comment (fraimondo) [reply junifer/utils/logging.py:95]: just copilot being copilot

File: junifer/utils/logging.py (hunk 1 above)
Comment (fraimondo) [reply junifer/utils/logging.py]: I will auto-detect color support

**Review by synchon — APPROVED** [review]
Needs a newsfragment and it's good to go.

**Pull request merged**